test/graaflib/algorithm/topological_sorting_test.cpp

```diff
@@ -0,0 +1,192 @@
+#include <fmt/core.h>
+#include <graaflib/algorithm/topological_sorting.h>
+#include <graaflib/graph.h>
+#include <graaflib/types.h>
+#include <gtest/gtest.h>
+
+namespace graaf::algorithm {
+namespace {
+template <typename T>
+struct TypedTopologicalSort : public testing::Test {
+  using graph_t = T;
+};
+
+using graph_types = testing::Types<directed_graph<int, int>>;
+TYPED_TEST_SUITE(TypedTopologicalSort, graph_types);
+
+};  // namespace
+
+TYPED_TEST(TypedTopologicalSort, ShortGraph) {
+  // GIVEN
+  using graph_t = typename TestFixture::graph_t;
+  graph_t graph{};
+
+  const auto vertex_1{graph.add_vertex(10)};
+  const auto vertex_2{graph.add_vertex(20)};
+  const auto vertex_3{graph.add_vertex(30)};
+  const auto vertex_4{graph.add_vertex(40)};
+
+  graph.add_edge(vertex_1, vertex_2, 25);
+  graph.add_edge(vertex_2, vertex_3, 35);
+  graph.add_edge(vertex_3, vertex_4, 45);
+
+  // WHEN;
+  auto sorted_vertices = topological_sort(graph);
+  std::vector<vertex_id_t> expected_vertices{0, 1, 2, 3};
```

Comment:
[minor]: I would propose to use the obtained vertex IDs in the `expected_vertices`. Then your tests are completely decoupled from the logic which generates vertex IDs.
```suggestion
  const auto vertex_1{graph.add_vertex(10)};
  const auto vertex_2{graph.add_vertex(20)};
  const auto vertex_3{graph.add_vertex(30)};
  const auto vertex_4{graph.add_vertex(40)};

  graph.add_edge(vertex_1, vertex_2, 25);
  graph.add_edge(vertex_2, vertex_3, 35);
  graph.add_edge(vertex_3, vertex_4, 45);

  // WHEN;
  auto sorted_vertices = topological_sort(graph);
  std::vector<vertex_id_t> expected_vertices{vertex_1, vertex_2, vertex_3, vertex_4};
```

Reply:
Same for the other tests